x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/schema/schema_logic.ts

```diff
@@ -349,7 +350,9 @@ export const SchemaLogic = kea<MakeLogicType<SchemaValues, SchemaActions>>({
       } catch (e) {
         window.scrollTo(0, 0);
         if (isAdding) {
-          actions.onSchemaSetFormErrors(e?.message);
+          // We expect body.message to be a string[] for actions.onSchemaSetFormErrors
+          const message: string[] = e?.body?.message || [defaultErrorMessage];
+          actions.onSchemaSetFormErrors(message);
         } else {
           flashAPIErrors(e);
         }
```

Comment:
Why not just add the `e?.body?.message` check into the conditional and let invalid messages fall back to `flashAPIErrors`, instead having to manually import `defaultErrorMessage`?

```suggestion
        const errorMessage: string[] = e?.body?.message;
        if (isAdding && errorMessage) {
          actions.onSchemaSetFormErrors(errorMessage);
        } else {
          flashAPIErrors(e);
        }
```

Reply:
I think this would be a slight functional difference, in that when the message is missing, it would appear as a Flash Message error.  [actions.onSchemaSetFormErrors sets addFormFieldErrors](https://github.com/elastic/kibana/blob/3a6f3d0f619644c48a1f45116882f8746a16322c/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/schema/schema_logic.ts#L207) which is [used inside a SchemaAddFieldModal](https://github.com/elastic/kibana/blob/3a6f3d0f619644c48a1f45116882f8746a16322c/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/schema/schema.tsx#L142).  I think we'd still want the default error message to appear in the SchemaAddFieldModal.

Reply:
I don't have a strong opinion either way. I guess I'm leaning slightly towards showing the error in the modal, closer to the user action that lead to the error. But I see the point in showing it as a flash message: if the error is unexpected, it's probably not caused by user action.

I will merge the PR as-is because the code fixes the original bug and is already tested.